### DISCUSSION TITLE
Should we release 0.66.4?

### DISCUSSION THREAD

**Discussion by lunaleaps** [reactwg/react-native-releases#6] [Patches]
## 0.66.4 has been released
See discussion about 0.66.5 here: https://github.com/reactwg/react-native-releases/discussions/8
----
Current Release: [0.66.3](https://github.com/facebook/react-native/releases/tag/v0.66.3)

Conversations on this thread are limited to 0.66 releases major issues and backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into main but did not make the 0.66.3 cut, with a link to the specific commit hash on main with the commit to cherry-pick, like this example link: facebook/react-native@bd2b7d6

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.

### List of cherry picks
1. facebook/react-native#32715
2. facebook/react-native#32714
### Local commits to backport to main
1.

**Comment by Yonom** [comment]
The `__apply_Xcode_12_5_M1_post_install_workaround` script causes problems if one has pods targetting `12.0` or higher. Can you consider picking this fix? https://github.com/facebook/react-native/commit/a4a3e675542827bb281a7ceccc7b8f5533eae29f

**Reply by lunaleaps** [reply]
Do you mind making a PR against the 0.66-stable branch?

**Reply by Yonom** [reply]
Done! https://github.com/facebook/react-native/pull/32715

**Comment by Yonom** [comment]
@mmmoussa [suggested](https://github.com/reactwg/react-native-releases/discussions/1#discussioncomment-1760809) picking this commit for rtl support: https://github.com/facebook/react-native/commit/70ddf46c8afcd720e188b6d82568eac6ac8125e6

**Reply by mmmoussa** [reply]
v0.66.0 broke rtl support on iOS, meaning our app's core experience is currently broken for all iOS rtl users.

**Reply by lunaleaps** [reply]
Do you mind making a PR against the 0.66-stable branch?

**Reply by mmmoussa** [reply]
https://github.com/facebook/react-native/pull/32714

**Comment by mmmoussa** [comment]
@lunaleaps is v0.66.4 planned for release soon?

**Reply by kelset** [reply]
Currently our main focus is making sure that 0.67 is ready to go - we haven't talked about 0.66.4 lately, sorry.

**Reply by mikehardy** [reply]
It's worth mentioning that you can do a local build off the 0.66-stable branch to base app development from while waiting for either 0.66.4 (unlikely) or 0.67 (probably just days away but you never know). I recognize that's maybe not comfortable but I also understand a hard-break on I18N when you have an app targeted for majority Arabic users, and want a fix process that's in your control. Local react-native build of 0.66-stable branch is the way to bring it under your control

**Reply by kelset** [reply]
we won't be doing 0.67.0 this year to avoid overloading anyone during the holidays, so it could be that we'll do a 0.66.4 with the couple of commits that were PRd & merged - I've asked and will let you know

**Reply by mmmoussa** [reply]
@mikehardy thanks for your suggestion, and @kelset thanks for checking.

**Reply by lunaleaps** [reply]
Hey guys, yes I was thinking of doing a 0.66.4 update given that 67 isn't going to be out this year. I wanted to wait until Wednesday/Thurs to create the build in case there were any other picks. Also consulting with others if it's a good idea.

Thanks for your understanding and sorry I should've communicated that better!

**Reply by lunaleaps** [reply]
Actually another thing, for your pick PRs, we should be creating commitlies on them so people can use those versions (more of a realization for myself than anything). However this might not be true for 0.66 (I can't recall when @hramos added commitly support) but note to self to check that pick PRs do build artifacts

**Reply by kelset** [reply]
@lunaleaps yeah I was thinking the same about commitlies but I checked the branch and the CI jobs and I couldn't see the one generating... but maybe I didn't look close enough 😓

**Reply by mmmoussa** [reply]
@lunaleaps will there be a 0.66.4 going out today or tomorrow?

**Comment by arunim2405** [comment]
If you guys do decide to release 0.66.4, can you guys please also include this [commit](https://github.com/facebook/react-native/commit/49a1460a379e3a71358fb38888477ce6ea17e81a). thanks for all the good work. 🙌

**Reply by lunaleaps** [reply]
Can you clarify why it's a blocking issue for you using 0.66 and also can you make a PR against 0.66-stable? Thanks!

**Reply by arunim2405** [reply]
This [pr](https://github.com/facebook/react-native/pull/31402) explains the use of the commit very well, will make a PR against 0.66-stable soon 👍

**Reply by arunim2405** [reply]
@lunaleaps https://github.com/facebook/react-native/pull/32725

**Comment by billnbell** [comment]
Getting split issues with Ruby... ?

```
Generating Pods project
[!] An error occurred while processing the post-install hook of the Podfile.

undefined method `split' for nil:NilClass

/Users/bill/theclub/TheClub/node_modules/react-native/scripts/react_native_pods.rb:241:in `block (2 levels) in __apply_Xcode_12_5_M1_post_install_workaround'
/Users/bill/theclub/TheClub/node_modules/react-native/scripts/react_native_pods.rb:239:in `each'
/Users/bill/theclub/TheClub/node_modules/react-native/scripts/react_native_pods.rb:239:in `block in __apply_Xcode_12_5_M1_post_install_workaround'
/Users/bill/theclub/TheClub/node_modules/react-native/scripts/react_native_pods.rb:238:in `each'
/Users/bill/theclub/TheClub/node_modules/react-native/scripts/react_native_pods.rb:238:in `__apply_Xcode_12_5_M1_post_install_workaround'
/Users/bill/theclub/TheClub/ios/Podfile:40:in `block (3 levels) in from_ruby'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-core-1.11.2/lib/cocoapods-core/podfile.rb:196:in `post_install!'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:945:in `run_podfile_post_install_hook'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:933:in `block in run_podfile_post_install_hooks'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/user_interface.rb:149:in `message'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:932:in `run_podfile_post_install_hooks'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:331:in `block (2 levels) in create_and_save_projects'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer/xcode/pods_project_generator/pods_project_writer.rb:61:in `write!'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:330:in `block in create_and_save_projects'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/user_interface.rb:64:in `section'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:309:in `create_and_save_projects'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:301:in `generate_pods_project'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:180:in `integrate'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:167:in `install!'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/command/install.rb:52:in `run'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/lib/cocoapods/command.rb:52:in `run'
/Users/bill/.rvm/gems/ruby-2.7.4/gems/cocoapods-1.11.2/bin/pod:55:in `<top (required)>'
/Users/bill/.rvm/gems/ruby-2.7.4/bin/pod:23:in `load'
/Users/bill/.rvm/gems/ruby-2.7.4/bin/pod:23:in `<main>'
/Users/bill/.rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `eval'
/Users/bill/.rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `<main>'
```

**Reply by billnbell** [reply]
If I remove split('.')... in node_modules/react-native/scripts/react_native_pods.rb to config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].to_f < 11.0 it works.

```
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      # ensure IPHONEOS_DEPLOYMENT_TARGET is at least 11.0
      should_upgrade = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].to_f < 11.0
      if should_upgrade
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
      end
    end
  end
```

**Reply by billnbell** [reply]
Patch file
```diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index df31139..13c6a3a 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -238,7 +238,7 @@ def __apply_Xcode_12_5_M1_post_install_workaround(installer)
   installer.pods_project.targets.each do |target|
     target.build_configurations.each do |config|
       # ensure IPHONEOS_DEPLOYMENT_TARGET is at least 11.0
-      should_upgrade = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].split('.')[0].to_i < 11
+      should_upgrade = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].to_f < 11.0
       if should_upgrade
         config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
       end
```

**Reply by mikehardy** [reply]
@Yonom oh no! :fearful:

**Reply by kelset** [reply]
@billnbell it seems like it might be related to some of the pods that you are using, could you share your Podfile here? Or, even better, create a simple repro?

**Reply by billnbell** [reply]
The new sample code works in cases where a POD does not include IPHONEOS_DEPLOYMENT_TARGET and I would recommend we move forward with the new patch.

**Reply by Yonom** [reply]
Whops, my bad 😞 

Something I worry about in the patch is that  `nil.to_f < 11.0 == true` and thus for pods that do not have `IPHONEOS_DEPLOYMENT_TARGET`, we end up setting it to 11.0. I am not sure if there are any side-effects of doing so... Perhaps its safer to skip them alltogether

**Reply by billnbell** [reply]
True you can check for 0.0 and if that is set ignore setting it. Good idea.

**Reply by billnbell** [reply]
Here is how you can do it

```
should_upgrade = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].to_f < 11.0 && config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].to_f != 0.0
```

**Reply by billnbell** [reply]
```
diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index df31139..71d1993 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -238,7 +238,7 @@ def __apply_Xcode_12_5_M1_post_install_workaround(installer)
   installer.pods_project.targets.each do |target|
     target.build_configurations.each do |config|
       # ensure IPHONEOS_DEPLOYMENT_TARGET is at least 11.0
-      should_upgrade = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].split('.')[0].to_i < 11
+      should_upgrade = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].to_f < 11.0 && config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'].to_f != 0.0
       if should_upgrade
         config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
       end
```

**Reply by billnbell** [reply]
I tested it on some bad pods, and it works great.

**Reply by kelset** [reply]
Can either of you submit a PR for this?

**Reply by Yonom** [reply]
@kelset https://github.com/facebook/react-native/pull/32746 (targetting main), https://github.com/facebook/react-native/pull/32747 (targetting 0.66-stable)

**Comment by billnbell** [comment]
ok can we release it?
